core/src/main/java/org/apache/brooklyn/util/core/flags/TypeCoercions.java

```diff
@@ -138,6 +139,9 @@ private TypeCoercions() {}
     @SuppressWarnings({ "unchecked" })
     public static <T> T coerce(Object value, TypeToken<T> targetTypeToken) {
         if (value==null) return null;
+        if (value instanceof DeferredSupplier<?>) {
+            value = ((DeferredSupplier<?>) value).get();
```

Comment:
@aledsage: Revisiting this, `DeferredSupplier` simply restates the contract of (Guava) `Supplier`. While superficially redundant, it may offer a (tenuous) distinction between "immediately resolvable" and "deferred-ly resolvable" quantities. Seems weak though, and potentially *very* confusing for users.